### PR TITLE
Integrated DynamicVariables from neeto-molecules

### LINKED ISSUE #679: Integrate `DynamicVariables` from neeto-molecules
Part of : https://github.com/bigbinary/neeto-molecules/issues/39

Replace existing custom implementation of `DynamicVariables` components using neeto-molecules

### PR DESCRIPTION
**Pull request by gaagul**
- Fixes #679 

**Description**

- Replaced: custom implementation of `DynamicVariables` using neeto-molecules

**Checklist**

- [x] I have made corresponding changes to the documentation.
~~- [ ] I have updated the types definition of modified exports.~~
- [x] I have verified the functionality in some of the neeto web-apps.
- [x] I have added the necessary label (patch/minor/major - If package publish
      is required).

**Reviewers**

@AbhayVAshokan _a

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -144,6 +145,7 @@
   },
   "peerDependencies": {
     "@bigbinary/neeto-commons-frontend": "latest",
+    "@bigbinary/neeto-molecules": "latest",
```
Comment (AbhayVAshokan) [inline package.json:148]: We should not add it to peer dependency before it added to `package-common.json` file. We'll make this change once that is done.

**Review by AbhayVAshokan — CHANGES_REQUESTED** [review]
- [x] Add a tooltip for the Variables button.
- [x] The button size should be "small".

@gaagul _a

**Comment by AbhayVAshokan** [comment]
@gaagul here is one additional change: https://vimeo.com/807511994/50142c0417

**Comment by gaagul** [comment]
@AbhayVAshokan _a Please review and verify the changes to the styling.

**Review by AbhayVAshokan — CHANGES_REQUESTED** [review]
@gaagul _a one more change: https://vimeo.com/807809112/3085c03e7f

**Comment by gaagul** [comment]
@AbhayVAshokan Please verify the new changes.

**Review by AbhayVAshokan — APPROVED** [review]

**Pull request merged**